pallets/ddc-clusters/src/lib.rs

```diff
@@ -821,6 +822,17 @@ pub mod pallet {
 			Ok(cluster.manager_id)
 		}
 
+		fn get_nodes(cluster_id: &ClusterId) -> Result<Vec<NodePubKey>, DispatchError> {
+			let mut nodes = Vec::new();
+
+			// Iterate through all nodes associated with the cluster_id
+			for (node_pubkey, _) in ClustersNodes::<T>::iter_prefix(cluster_id) {
```

Comment:
Stuff like this is only applicable in scope of demo. Having unbounded iterables in the runtime code is a security hole that can used by a malicious actor to affect the network bandwidth and even fully block it.